sdk/python/tests/test_feature_store.py

```diff
@@ -203,27 +207,46 @@ def test_feature_view_inference_success(test_feature_store, dataframe_source):
             entities=["id"],
             ttl=timedelta(minutes=5),
             online=True,
-            input=prep_bq_source(dataframe_source, "ts_1"),
+            input=simple_bq_source_using_table_ref_arg(dataframe_source, "ts_1"),
+            tags={},
+        )
+
+        fv3 = FeatureView(
+            name="fv3",
+            entities=["id"],
+            ttl=timedelta(minutes=5),
+            online=True,
+            input=simple_bq_source_using_query_arg(dataframe_source, "ts_1"),
             tags={},
         )
 
-        test_feature_store.apply([fv1, fv2])  # Register Feature Views
+        test_feature_store.apply([fv1, fv2, fv3])  # Register Feature Views
         feature_view_1 = test_feature_store.list_feature_views()[0]
         feature_view_2 = test_feature_store.list_feature_views()[1]
+        feature_view_3 = test_feature_store.list_feature_views()[2]
 
         actual_file_source = {
             (feature.name, feature.dtype) for feature in feature_view_1.features
         }
-        actual_bq_source = {
+        actual_bq_using_table_ref_arg_source = {
             (feature.name, feature.dtype) for feature in feature_view_2.features
         }
+        actual_bq_using_query_arg_source = {
+            (feature.name, feature.dtype) for feature in feature_view_3.features
+        }
         expected = {
             ("float_col", ValueType.DOUBLE),
             ("int64_col", ValueType.INT64),
             ("string_col", ValueType.STRING),
         }
+        expected2 = {  # parsing with the "query" param in bq sources uses different code path
```

Comment:
I think we want both code paths to end up producing the same types, see also comment on types above